src/services/report.ts

```diff
@@ -65,5 +65,17 @@ class Report {
     };
     xhr.send(JSON.stringify(data));
   }
+
+  public sendByBeacon(data: any) {
+    if (!this.url) {
+      return;
+    }
+    if (typeof navigator.sendBeacon === 'function') {
+      navigator.sendBeacon(this.url, JSON.stringify(data));
```

Comment:
Protocols now don't support to send string data. https://github.com/apache/skywalking/blob/master/docs/en/protocols/Browser-HTTP-API-Protocol.md

Reply:
ok, i will change string data to URLSearchParams.

Reply:
> Protocols now don't support to send string data. https://github.com/apache/skywalking/blob/master/docs/en/protocols/Browser-HTTP-API-Protocol.md

I found that no data can suport to send a array data.So shell i waiting for you to modify protocols to an object instead of  an array ?

Reply:
What kind of an array are you asking?

Reply:
> What kind of an array are you asking?

![image](https://user-images.githubusercontent.com/21191777/169763900-13ce3e22-438a-49a1-a688-bc22ee22c325.png)

Reply:
I don't think we had this ever. What is changing right now?

Reply:
> I don't think we had this ever. What is changing right now?

I mean that your transfer data with data string.But sendBeacon suport a ArrayBuffer, ArrayBufferView, String, FormData, or URLSearchParams object.Now your protocal is an data string, not a object.Just like Fine0830 said. Did i say clear?

Reply:
Hang back, let me check it out.

Reply:
I need to use [this](https://github.com/SkyAPMTest/skywalking-client-test) to test and validate this pr, but the project now doesn't work. Waiting for @arugal checking it out. BTW, I think our protocols can support this with String data and the `sendBeacon` also support send String data, just like using the `XMLHttpRequest` object. That [doc](https://github.com/apache/skywalking/blob/master/docs/en/protocols/Browser-HTTP-API-Protocol.md) only explains the reported data fields.